src/latexify/codegen/function_codegen.py

```diff
@@ -619,13 +619,17 @@ def _get_sum_prod_range(self, node: ast.comprehension) -> tuple[str, str] | None
             lower_rhs = f"{{{range_info.start_int}}}"
 
         if range_info.stop_int is None:
-            # use special processing if range_info.stop involves addition or subtraction
-            if isinstance(range_info.stop, ast.BinOp) and isinstance(
-                range_info.stop.op, (ast.Add, ast.Sub)
+            stop = range_info.stop
+            # use special processing if stop is infinity
+            if isinstance(range_info.stop, ast.Name) and str(stop.id) == "infty":
+                upper = "{" + self.visit(stop) + "}"
```

Comment:
This is always applied regardless of the `use_math_symbols` option. It would be better to maintain the original behavior if the flag is off.

Reply:
fixed

Reply:
It is recommended to check the flag directly rather than checking the result of visit since visit is not stable (as it is affected by any flags).